src/main/java/au/org/ala/biocache/service/ListsService.java

```diff
@@ -300,7 +300,7 @@ public Kvp find(List<Kvp> kvps, Kvp lftrgt) {
             return kvps.get(idx);
         } else {
             // reverse through kvps until a match is found
-            idx = idx * -1;
+            idx = Math.min(idx * -1, kvps.size() -1);
```

Comment:
Based on the `Collections.binarySearch` documentation https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#binarySearch(java.util.List,%20T,%20java.util.Comparator):

If the exact binary search failed it is defined to return the index `(- (insertion point) -1)`  .

If the element belongs at the end of the list it seems like `insertion point` may be `kvps.size()`, and `idx` will be `-kvps.size() - 1`. The issue in that case seems to be in our use of `idx * -1` to change that to a positive index, as it will end up being `kvps.size() + 1` which is an invalid list index.

This fix seems to cater for the case where `idx * -1` ends up being `kvps.size() + 1` by changing it to `kvps.size() - 1` which is the top index in the list. It also changes the case where `idx * -1` ends up being `kvps.size()` which is also an invalid element index.